### ISSUE TITLE
Render a Website to Canvas UI

### ISSUE THREAD

**Issue by fngryboi** [uWebKit/uWebKit#11]
Currently this seems to be impossible to pull off, as none of the example demonstrates displaying a website onto a Canvas...

UPDATE: I've played around a bit with the scripts and I'm working on a fork of WebTexture. Will post the results here when it's working 100% (Currently only able to render a webpage but no raycasting support as of yet).

**Comment by fngryboi** [comment]
Scene Setup:
* Add a Canvas
* Add a Image Object inside the Canvas
* Add the tag "WebView" to the Image Object
* Add the Script UWKWebView and set it up
* Add the Script below:
* Create a empty material and add it to the WebUI script material

Here's the code:

`/******************************************
  * uWebKit 
  * (c) 2014 THUNDERBEAST GAMES, LLC
  * http://www.uwebkit.com
  * sales@uwebkit.com
  * 
  * UI fork made by @fngryboi based on uWebkitExamples/Scripts/WebTexture.cs
*******************************************/

using UnityEngine;
using UnityEngine.UI;
using System.Collections;
using System.Collections.Generic;
using UnityEngine.EventSystems;

/// <summary>
/// Basic example of using a UWKWebView on a 3D Unity UI Component
/// </summary>


public class WebUI : MonoBehaviour
{

	#region Inspector Fields
	public bool KeyboardEnabled = true;
	public bool MouseEnabled = true;
	public bool HasFocus = true;
	public bool AlphaMask = false;
	#endregion

	public Image ImageSrc;

	public Material materialSrc;

	UWKWebView view;

	// Use this for initialization
	void Start ()
	{   

		view = gameObject.GetComponent<UWKWebView>();

		//view.SetAlphaMask(AlphaMask);

		//Assign the specific Image you want
		ImageSrc = GetComponent<Image>();

		//Instantiate a new material and append the texture to the material before adding it to the Image component. If you just throw a material to an image and you have several image component in the same canvas they will all get the same image material. (Bug in Unity 5.3.4f1, reported)
		//Might be useful with keeping several tabs/windows open aswell - need further iteration for this to work
		Material webView = Instantiate (materialSrc);
		webView.mainTexture = view.WebTexture;

		ImageSrc.material = webView;
	}

	// Update is called once per frame
	void Update ()
	{

		if (!MouseEnabled || !HasFocus)
			return;

		else {
			PointerEventData pointer = new PointerEventData(EventSystem.current);
			pointer.position = Input.mousePosition;

			List<RaycastResult> raycastResults = new List<RaycastResult>();
			EventSystem.current.RaycastAll(pointer, raycastResults);

			if(raycastResults.Count > 0)
			{
				if (raycastResults [0].gameObject.tag == "WebView") {	

					//Quick and dirty, should be 1 to 1 in a Screen Overlay scenario. Won't work very well in World Space scenarios so this part will need further iteration!

					int x = (int)(pointer.position.x * (float)view.MaxWidth) / Screen.width;
					int y = view.MaxHeight - (int)(pointer.position.y * (float)view.MaxHeight) / Screen.height;

					Debug.Log("Mouse position: " + x + " _ " + y);

					Vector3 mousePos = new Vector3();
					mousePos.x = x; 
					mousePos.y = y;
					view.ProcessMouse(mousePos); 
				}
			}
		}
	}

	void OnGUI ()
	{       
		if (!KeyboardEnabled || !HasFocus)
			return;

		if (Event.current.isKey)
		{
			view.ProcessKeyboard(Event.current);
		}

	}

}`

**Comment by JoshEngebretson** [comment]
Cool!  Thanks for posting the script.  I will reopen this issue as we do need to include an example

**Comment by fngryboi** [comment]
Yeah no problem, the mouse position translation might be a little bit off as I was going to use an external asset called CurvedUI, which have their own Raycaster script included.
I found Canvas to be a bit wonky when used with a Raycaster when the UI is in world space... I think it gave the mouse position as the position on the canvas (related to its size), making it difficult when you have a canvas of 160*90 but the WebView is 1920*1080. In other words the mouse position only goes between 0,0 and 160,90.

A good way to solve it would be to have the canvas in the same size as the WebView is rendering the webpage. Or bigger and then divide the x and y position of the mouse relative to the ratio between the webpage dimensions and the canvas dimensions.
